bottles/backend/health.py

```diff
@@ -104,38 +93,33 @@ def check_x11(self):
 
     @staticmethod
     def check_wayland():
-        if "WAYLAND_DISPLAY" in os.environ:
-            return True
-        return False
+        return "WAYLAND_DISPLAY" in os.environ
 
     def check_xwayland(self):
-        if self.x11 and self.wayland:
-            return True
-        return False
+        return self.x11 and self.wayland
 
     def check_desktop(self):
-        return os.environ.get("DESKTOP_SESSION")
+        return os.environ.get("XDG_CURRENT_DESKTOP")
 
     @staticmethod
     def check_cabextract():
-        res = shutil.which("cabextract")
-        if res is None:
-            return False
-        return True
+        return bool(shutil.which("cabextract"))
 
     @staticmethod
     def check_p7zip():
-        res = shutil.which("7z")
-        if res is None:
-            return False
-        return True
+        return bool(shutil.which("7z"))
 
     @staticmethod
     def check_patool():
-        res = shutil.which("patool")
-        if res is None:
-            return False
-        return True
+        return bool(shutil.which("patool"))
+
+    @staticmethod
+    def check_xdpyinfo():
+        return bool(shutil.which("xdpyinfo"))
+
+    @staticmethod
+    def check_ImageMagick():
+        return bool(shutil.which("identify"))
 
     @staticmethod
     def check_icoextract():
```

Comment:
Many other `check_*` methods here are just a `try`, `import`, and a `catch`. Is there no way to make it nicer, and maybe take advantage of Python's dynamic type system here to convert any exceptions to `False`? 

It also looks like mypy might not be getting applied to these files; they don't have a type declaration AIUI.

Reply:
An alternative method, courtesy of @refi64, because apparently an Exception -> bool conversion ain't happening:

```python
def check_module(module):
  try:
    __import__(module)
  except ModuleNotFoundError:
    return False
  else:
    return True
    ```
    
    And we can get rid of many of these individual `check_*` methods.

Reply:
And a similar cleanup for all the `schutil` methods.

Reply:
That indeed would be much cleaner, I'll implement the changes.

Reply:
We can get rid of each of the individual methods too, not just redirect them.

Reply:
I have no idea of what you have in mind. I'll close the PR but keep the branch intact so you can still work on it.

Reply:
I mean, instead of `check_*`, we do `check_module()` and `check_executable()`, but iterate over the names and whatnot. Thanks for kicking off most of this again anyways; everything else but what I'm talking about can be merged now, if you'd like to reopen and remove the changes.